pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -330,37 +342,11 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		return nil
 	}
 
-	pipelineMeta, pipelineSpec, err := resources.GetPipelineData(ctx, pr, getPipelineFunc)
-	if err != nil {
-		logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
-		pr.Status.MarkFailed(ReasonCouldntGetPipeline,
-			"Error retrieving pipeline for pipelinerun %s/%s: %s",
-			pr.Namespace, pr.Name, err)
-		return controller.NewPermanentError(err)
-	}
-
 	// Store the fetched PipelineSpec on the PipelineRun for auditing
 	if err := storePipelineSpec(ctx, pr, pipelineSpec); err != nil {
```

Comment:
I think it might be a good idea to move this up along with `GetPipelineData`. Otherwise a Pending PipelineRun will end up fetching the data multiple times from the cluster / bundle: first while it's in Pending state and then again when it's moved out from Pending and reaches here and stores the spec in the status. wdyt?

Reply:
makes sense, moved it up, thanks!